File: Leetcode/Java/Q70_Climbing_Stairs.java
class Solution {
    public int climbStairs(int n) {
        // Create an array dp where dp[i] will store the number of ways to reach the ith step
        int[] dp = new int[n + 1];
        
        // There is one way to stay at the ground (step 0), by doing nothing
        dp[0] = 1;
        
        // There is one way to reach the first step, by taking one step
        dp[1] = 1;
        
        // Iterate from step 2 to step n to fill the dp array
        for (int i = 2; i <= n; i++) {
            // The number of ways to reach the ith step is the sum of ways to reach the (i-1)th step and (i-2)th step
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        
        // Return the number of ways to reach the nth step
        return dp[n];
    }
}

File: Leetcode/Python/Q70_Climbing_Stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        # Initialize a list to store the number of ways to reach each step
        dp = []
        
        # Base cases: there is 1 way to climb 0 steps and 1 way to climb 1 step
        dp.append(1)  # dp[0] = 1 (1 way to stay at the ground)
        dp.append(1)  # dp[1] = 1 (1 way to climb 1 step)

        # Compute the number of ways to climb to each step from 2 up to n
        # dp[i] is the sum of the ways to climb to the two previous steps (dp[i-1] + dp[i-2])
        for i in range(2, n + 1):
            dp.append(dp[i-1] + dp[i-2])

        # Return the number of ways to climb n steps
        return dp[n]
